### DISCUSSION TITLE
xdg-desktop

### DISCUSSION THREAD

**Discussion by DasLixou** [qtile/qtile#3896] [Q&A]
so i installed xdg-desktop because of obs-studio-git and now firefox starts slowly. found out thats because XDG_CURRENT_DESKTOP isnt set. so i made XDG_CURRENT_DESKTOP=QTile in /etc/enviroment. now its super weird because i have to startx, start firefox, then stop qtile, then restart startx to make it work. Anyone know how to fix? because internet says wm should set xdg session env variables, but it doesnt

**Comment by m-col** [comment]
You can stick `export XDG_CURRENT_DESKTOP=qtile` in your `~/.xinitrc` script before qtile starts

**Reply by m-col** [reply]
For further info regarding whether qtile should be setting this (which I don't think it should), see https://github.com/swaywm/sway/pull/4876 for a discussion (in the context of wayland), and i3, which doesn't set it for x11. Tldr some clients use it incorrectly, and different users want different values (you might want to set it to XFCE even in a qtile session to get certain GTK3 effects, for example).
